projects/wave-core-new/src/lib/project/project.service.ts

```diff
@@ -1045,54 +1044,65 @@ export class ProjectService {
-    //         this.getTimeStream(),
-    //         observableCombineLatest([
-    //             this.getProjectionStream(),
-    //             this.mapService.getViewportSizeStream()
-    //         ]).pipe(
-    //             debounceTime(this.config.DELAYS.DEBOUNCE)
-    //         ),
-    //     ]).pipe(
-    //         tap(() => loadingState$.next(LoadingState.LOADING)),
-    //         switchMap(([time, [projection, viewportSize]]) => {
-    //             const requestExtent: [number, number, number, number] = [0, 0, 0, 0];
-    //
-    //             let clusteredOption;
-    //             // TODO: is clustering a property of a layer or the symbology?
-    //             if (layer.clustered && layer.symbology instanceof PointSymbology) {
-    //                 clusteredOption = {
-    //                     minRadius: layer.symbology.radius,
-    //                 };
-    //             }
-    //
-    //             return this.mappingQueryService.getWFSData({
-    //                 operator: layer.operator,
-    //                 time,
-    //                 projection,
-    //                 clusteredOption,
-    //                 outputFormat: WFSOutputFormats.JSON,
-    //                 viewportSize
-    //             }).pipe(map(x => VectorData.olParse(time, projection, requestExtent, x)));
-    //         }),
-    //         tap(
-    //             () => loadingState$.next(LoadingState.OK),
-    //             (reason: Response) => {
-    //                 this.notificationService.error(`${layer.name}: ${reason.status} ${reason.statusText}`);
-    //                 loadingState$.next(LoadingState.ERROR);
-    //             }
-    //         ),
-    //     ).subscribe(
-    //         data => data$.next(data),
-    //         error => error // ignore error
-    //     );
-    // }
-    //
+    /**
+     * Create a subscription for layer data, symbology and provenance with loading state checks and error handling
+     */
+    private createVectorLayerDataSubscription(layer: VectorLayer,
+                                              data$: Observer<VectorData>,
+                                              loadingState$: Observer<LoadingState>): Subscription {
+        return combineLatest([
+            this.getTimeStream(),
+            combineLatest([
+                this.getProjectionStream(),
+                this.mapService.getViewportSizeStream()
+            ]).pipe(
+                debounceTime(this.config.DELAYS.DEBOUNCE)
+            ),
+            this.userService.getSessionTokenForRequest(),
+        ]).pipe(
+            tap(() => loadingState$.next(LoadingState.LOADING)),
+            switchMap(([time, [projection, viewportSize], sessionToken]) => {
+                const requestExtent: [number, number, number, number] = [0, 0, 0, 0];
+
+                // let clusteredOption;
+                // TODO: is clustering a property of a layer or the symbology?
```

Comment:
i think clustering is an operator. However, users want to enable this via the symbology-editor.